graph/src/env.rs

```diff
@@ -53,3 +64,955 @@ pub fn env_var<E: std::error::Error + Send + Sync, T: FromStr<Err = E> + Eq>(
+    #[envconfig(from = "GRAPH_EXPLORER_QUERY_THRESHOLD", default = "500")]
+    explorer_query_threshold_in_msec: u64,
+
+    // 1MiB
+    #[envconfig(from = "GRAPH_MAX_IPFS_CACHE_FILE_SIZE", default = "1048576")]
+    max_ipfs_cache_file_size: usize,
+    #[envconfig(from = "GRAPH_MAX_IPFS_CACHE_SIZE", default = "50")]
+    max_ipfs_cache_size: u64,
+    #[envconfig(from = "GRAPH_IPFS_TIMEOUT", default = "30")]
+    ipfs_timeout_in_secs: u64,
+    // 256MiB
+    #[envconfig(from = "GRAPH_MAX_IPFS_MAP_FILE_SIZE", default = "268435456")]
+    max_ipfs_map_file_size: usize,
+    #[envconfig(from = "GRAPH_MAX_IPFS_FILE_BYTES")]
+    max_ipfs_file_bytes: Option<usize>,
+
+    #[envconfig(from = "ETHEREUM_REORG_THRESHOLD", default = "250")]
+    ethereum_reorg_threshold: BlockNumber,
+    #[envconfig(from = "ETHEREUM_TRACE_STREAM_STEP_SIZE", default = "50")]
+    ethereum_trace_stream_step_size: BlockNumber,
+    #[envconfig(from = "GRAPH_ETHEREUM_MAX_EVENT_ONLY_RANGE", default = "500")]
+    ethereum_max_event_only_range: BlockNumber,
+    #[envconfig(from = "ETHEREUM_BLOCK_BATCH_SIZE", default = "10")]
+    ethereum_block_batch_size: usize,
+    #[envconfig(from = "GRAPH_ETHEREUM_JSON_RPC_TIMEOUT", default = "180")]
+    ethereum_json_rpc_timeout_in_secs: u64,
+    #[envconfig(from = "GRAPH_ETHEREUM_REQUEST_RETRIES", default = "10")]
+    ethereum_request_retries: usize,
+    #[envconfig(from = "GRAPH_GETH_ETH_CALL_ERRORS", default = "")]
+    geth_eth_call_errors: String,
+    #[envconfig(
+        from = "GRAPH_ETHEREUM_BLOCK_INGESTOR_MAX_CONCURRENT_JSON_RPC_CALLS_FOR_TXN_RECEIPTS",
+        default = "1000"
+    )]
+    ethereum_block_ingestor_max_concurrent_json_rpc_calls: usize,
+    #[envconfig(from = "GRAPH_ETHEREUM_FETCH_TXN_RECEIPTS_IN_BATCHES")]
+    ethereum_fetch_receipts_in_batches: Option<EnvVarBoolean>,
+    #[envconfig(from = "GRAPH_ETHEREUM_CLEANUP_BLOCKS", default = "false")]
+    ethereum_cleanup_blocks: EnvVarBoolean,
+
+    #[envconfig(from = "GRAPH_GRAPHQL_QUERY_TIMEOUT")]
+    graphql_query_timeout_in_secs: Option<u64>,
+    #[envconfig(from = "GRAPH_GRAPHQL_MAX_COMPLEXITY")]
+    graphql_max_complexity: Option<WithoutUnderscores<u64>>,
+    // u8::MAX
+    #[envconfig(from = "GRAPH_GRAPHQL_MAX_DEPTH", default = "255")]
+    graphql_max_depth: u8,
+    #[envconfig(from = "GRAPH_GRAPHQL_MAX_FIRST", default = "1000")]
+    graphql_max_first: u32,
+    // u32::MAX
+    #[envconfig(from = "GRAPH_GRAPHQL_MAX_SKIP", default = "4294967295")]
+    graphql_max_skip: u32,
+    #[envconfig(from = "GRAPHQL_ALLOW_DEPLOYMENT_CHANGE", default = "false")]
+    graphql_allow_deployment_change: EnvVarBoolean,
+    // usize::MAX
+    #[envconfig(
+        from = "GRAPH_GRAPHQL_WARN_RESULT_SIZE",
+        default = "18446744073709551615"
+    )]
+    graphql_warn_result_size: WithoutUnderscores<usize>,
+    // usize::MAX
+    #[envconfig(
+        from = "GRAPH_GRAPHQL_ERROR_RESULT_SIZE",
+        default = "18446744073709551615"
+    )]
```

Comment:
Unfortunately, it's doesn't. `envy` only supports symbols, meaning we can use default functions and constants, while `envconfig` only supports literal. Once `envy` supports both (https://github.com/serde-rs/serde/issues/368), we should 100% switch to `envy` imo.

Reply:
Personally I'd rather write `default = "ten_thousand"` than `default = "18446744073709551615"`. But it's a bikeshed. @mangas do you want to cast the deciding vote?

Reply:
I would definitely prefer not to hardcode the value, especially because usize is platform dependent. We could work around this implementing a new type with FromStr which could basically parse this and either panic or default to usize::MAX

Reply:
We have no control over `FromStr` if the environment variable is not present at all. We must provide a literal default if we stick with `envconfig`.

Reply:
In that case default could be "" and it would be handled in the FromStr implementation tough

Reply:
Uuuh, I see what you mean now. We can have something like a `struct ParseOrElse<T: FromStr>(T)` and it can work. Really nice!

Reply:
Fixed in https://github.com/graphprotocol/graph-node/pull/3364/commits/203d4340dcb06771202c0e48cd75b0fc450ae52b, marking as resolved.